scipy/stats/_continuous_distns.py

```diff
@@ -661,6 +661,62 @@ def _munp(self, n, c, d):
 burr = burr_gen(a=0.0, name='burr')
 
 
+class burr12_gen(rv_continuous):
+    """A Burr (Type XII) continuous random variable.
+
+    %(before_notes)s
+
+    See Also
+    --------
+    fisk : a special case of `burr` with ``d = 1``
+
+    Notes
+    -----
+    The probability density function for `burr` is::
+
+        burr.pdf(x, c, d) = c * d * x**(c-1) * (1+x**(c))**(-d-1)
+
+    for ``x > 0``.
+
+    `burr` takes ``c`` and ``d`` as shape parameters.
+
+    This is the PDF corresponding to the twelfth CDF given in Burr's list;
+    specifically, it is equation (20) in Burr's paper [1]_.
+
+    %(after_notes)s 
+    
+    The Burr type 12 distribution is also sometimes referred to as
+    the Singh-Maddala distribution" from NIST
+
+    References
+    ----------
+    .. [1] Burr, I. W. "Cumulative frequency functions", Annals of
+       Mathematical Statistics, 13(2), pp 215-232 (1942).
+
+    %(example)s
+
+    """
+    def _pdf(self, x, c, d):
+        return c*d*(x**(c-1.0))*((1+x**(c*1.0))**(-d-1.0))
+        
+    def _logpdf(self, x, c, d):
+        return log(c) + log(d) + special.xlogy(c-1, x) + special.xlog1py(-d-1, x**c)
+
+    def _cdf(self, x, c, d):
+        return (1-(1+x**(c*1.0))**(-d**1.0))
```

Comment:
Oh, I used the floating point values to ensure that even in case of integer values of ``c`` and ``d`` would be still floating numbers. I have seen the use of raising to power of 1.0 in some functions in ``_continuous_distns.py``, including the similarly defined ``burr( type 3)``. Will remove them if that is unnecessary.